### PR TITLE
fix: use new TS API

### LINKED ISSUE #1: [Default] AttributeError: type object 'tree_sitter.Language' has no attribute 'build_library'
Binary Ninja traceback:

```python3
[Default] Traceback (most recent call last):
[Default]   File "/Applications/Binary Ninja.app/Contents/MacOS/plugins/../../Resources/python/binaryninja/plugin.py", line 127, in _default_action
[Default]     action(view_obj)
[Default]   File "/Users//user/Library/Application Support/Binary Ninja/plugins/semgrep_bn/__init__.py", line 462, in run
[Default]     SemgrepAnalysis(bv).start()
[Default]   File "/Users/user/Library/Application Support/Binary Ninja/plugins/semgrep_bn/__init__.py", line 254, in __init__
[Default]     Language.build_library(str(TREE_SITTER_LIB), [str(TREE_SITTER_C)])
[Default] AttributeError: type object 'tree_sitter.Language' has no attribute 'build_library'
```

Python 3.11 autocomplete suggests that build_library was removed from the .Language member.
```python3
Python 3.11.3 (v3.11.3:f3909b8bc8, Apr  4 2023, 20:12:10) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tree_sitter
>>> tree_sitter.Language.
tree_sitter.Language.field_count
tree_sitter.Language.field_id_for_name(
tree_sitter.Language.field_name_for_id(
tree_sitter.Language.id_for_node_kind(
tree_sitter.Language.lookahead_iterator(
tree_sitter.Language.mro()
tree_sitter.Language.next_state(
tree_sitter.Language.node_kind_count
tree_sitter.Language.node_kind_for_id(
tree_sitter.Language.node_kind_is_named(
tree_sitter.Language.node_kind_is_visible(
tree_sitter.Language.parse_state_count
tree_sitter.Language.query(
tree_sitter.Language.version
```

### PR DESCRIPTION
**Pull request by 0xricksanchez**
Fixes #1 

TLDR: Keeping up with breaking tree-sitter changes is hard.

PS: Currently the plugin is broken and cannot be used. Alternatively, pin the `tree-sitter` package to `v0.21.3`

### CODE REVIEW

**Comment by adrianherrera** [comment]
Sorry, I only jus saw this PR! Let me give it a try and see how it goes

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 emoji
 tree-sitter
-semgrep
\ No newline at end of file
+tree-sitter-c
```
Comment (adrianherrera) [inline requirements.txt]: Given the proclivity of tree-sitter API updates, maybe it's worth pinning these like you said?

When I tested your PR I had

```
tree-sitter==0.23.2
tree-sitter-c==0.23.4
```

File: __init__.py [hunk 2]
```diff
@@ -443,13 +429,16 @@ def remove_function_annotations(self, src: str) -> str:
         rest is kept the same.
         """
         tree = self.parser.parse(bytes(src, "utf8"))
+        # captures now returns a dict of capture name -> list of nodes
```
Comment (adrianherrera) [inline __init__.py]: This comment probably isn't needed. Just delete, and we'll pin in requirements.txt

**Comment by adrianherrera** [comment]
Ok it looks good! Just a few comments to address, then I'm happy to merge. Sorry again for the delay.

**Comment by 0xricksanchez** [comment]
@adrianherrera done

**Comment by adrianherrera** [comment]
all merged! Thanks again for the PR.
